### ISSUE TITLE
[BUG/ISSUE] Problem running Hg in GC 13.3.1

### ISSUE THREAD

**Issue by Jampo26** [geoschem/geos-chem#1011]
### Description of the problem
Hello. We are using GC v13.3.1 to test the mercury module, but have encountered some problems as follows:
In the step of Creating run directories, we choose the setting of MERRA-2, 4°×5°, and 47 levels. After compiling the model successfully, we did a dry-run for 20190101-20200101, and downloaded the input data we need. After a 1-year running, we found the distribution of Hg0 concentration and some other dignostic results are abnormal. 

### Description of troubleshooting performed
Although the geos.log shows the WHET files are opened, but we check the output files "GEOSChem.MercuryEmis.2019****_0000z.nc4" and find the values of several variables are zero (e.g. EmisHg0anthro, EmisHg0geogenic ...). The defalut anthropogenic Hg emissions using WHET 2010 (this part is true in HEMCO_Config.rc), but it seems that the WHET emissions are not included in the running. We wonder if there is a bug in v13.3.1 when reading in emissions of Hg?
We are now running a new test from 2015 (see geos.log), but the emissions are still missing. The related settings and some results (year 2019) are attached below, please check it out.

### GEOS-Chem version
GC 13.3.1

### Description of modifications
No substantial code changes.

### Log files
- Build log (if applicable):
[input.geos.txt](https://github.com/geoschem/geos-chem/files/7539594/input.geos.txt)
[HEMCO_Config.rc.txt](https://github.com/geoschem/geos-chem/files/7539595/HEMCO_Config.rc.txt)

- Run logs (if applicable):
[geos.log](https://github.com/geoschem/geos-chem/files/7539570/geos.log)

- Run script (if applicable):

### Some results
[results.zip](https://github.com/geoschem/geos-chem/files/7539748/results.zip)

### Software versions
- CMake version: 3.19.0
- Compilers (Intel or GNU, and version): intel2017u1
- NetCDF version: 4.5.0

**Comment by yantosca** [comment]
Thanks for writing @Jampo26.  To my knowledge, the Hg simulation diagnostics have never really been validated well.

I might suggest trying to add entries to HEMCO_Diagn.rc so as to print out the Hg emissions.  That might be quicker than trying to debug the current situation.

**Comment by yantosca** [comment]
Also I think part of the issue is that the Hg emissions diagnostics were implemented as HEMCO manual diagnostics, in this routine:

https://github.com/geoschem/geos-chem/blob/e9a4f1f9873f6e12ef3b9af7f823368a97339aa7/GeosCore/hcoi_gc_diagn_mod.F90#L864-L1070

But with GEOS-Chem having been updated frequently and the Hg code having lied kind of stale, it is possible that something got broken along the way.

**Comment by Jampo26** [comment]
Thanks for your reply @yantosca. I did the settings of HEMCO_Diagn in HEMCO_Config.rc and got "HEMCO_diagnostics.****.nc" in the output files. The values of **Hg0_emission_flux_from_anthropogenic** and other variables don't seem to have any problem, but the distribution of Hg0 concentrations in the GEOSChem.SpeciesConc.****.nc4 is still abnormal. Seems like the anthropogenic emissions are not really read into the model？
**Log files**
[HEMCO_Diagn.rc.txt](https://github.com/geoschem/geos-chem/files/7544085/HEMCO_Diagn.rc.txt)
[geos.log2.txt](https://github.com/geoschem/geos-chem/files/7544097/geos.log2.txt)

**Comment by yantosca** [comment]
@msl3v or @viral211, would either of you be able to look at this?  Thanks.

**Comment by viral211** [comment]
It seems to me too that Hg emissions are not being processed correctly. This is not an issue with the Hg diagnostics. In mercury_mod.F90/mercury_readyr, emissions are fed in through the HEMCO diagnostics that are defined in GeosCore/hcoi_gc_diagn_mod.F90/Diagn_Hg. There seems to be a problem with updating these diagnostic fields in HEMCO. When trying to read the Hg0_Anthro diagnostic, for example, the HEMCO.log file shows the following warning:
```
HEMCO WARNING: Diagnostics counter is zero - return empty array: HG0_ANTHRO
--> LOCATION: DiagnCont_PrepareOutput (hco_diagn_mod.F90)
```
Could @jimmielin help with this? thanks.

**Comment by jimmielin** [comment]
Hi all,

This sounds like an issue from `mercury_mod.F90`'s reading of HEMCO diagnostics to feed emissions to `mercury_readyr`. I am unfamiliar with `mercury_mod` but I remember that `Diagnostics counter is zero` usually means that [a call to `Diagn_Update` is missing](http://wiki.seas.harvard.edu/geos-chem/index.php/MEGAN_v2.1_plus_Guenther_2012_biogenic_emissions). Has the code been changed recently? I can't find that call in `mercury_mod`.

Best,
Haipeng

**Comment by viral211** [comment]
Thanks @jimmielin. No, there has been no recent change to `mercury_mod`, other than the general model structural updates. `Diagn_Update` was never called from `mercrury_mod`.

**Comment by yantosca** [comment]
Thanks @jimmielin.  The Hg code has been a bit neglected.  I may not be able to look into this for a while but we do have the Hg code update for KPP coming soon.

**Comment by yantosca** [comment]
This was also noted in #895

**Comment by msulprizio** [comment]
I believe the fix submitted in https://github.com/geoschem/HEMCO/pull/121 should resolve this issue. I have asked @yantosca to confirm that this does in fact resolve the diagnostic issues in the Hg simulation.

**Comment by yantosca** [comment]
I confirm that the fix in https://github.com/geoschem/HEMCO/pull/121 does result in Hg emissions being added to the SpeciesConc and MercuryEmis collections.  I will release a patch version GEOS-Chem 13.3.3.

**Comment by yantosca** [comment]
This should now be fixed in [GEOS-Chem 13.3.3](https://github.com/geoschem/GCClassic/releases/tag/13.3.3).  We can close out this issue.
